src/Diagnostics.DataProviders/ChangeAnalysisClient.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Diagnostics.DataProviders.Interfaces;
+using Newtonsoft.Json;
+using Diagnostics.ModelsAndUtils.Models.ChangeAnalysis;
+using Diagnostics.DataProviders.TokenService;
+using System.Collections.Generic;
+using System.Net;
+
+namespace Diagnostics.DataProviders
+{
+    public class ChangeAnalysisClient : IChangeAnalysisClient
+    {
+        /// <summary>
+        /// x-ms-client-object-id header to pass to Change Analysis endpoint.
+        /// </summary>
+        private string clientObjectIdHeader;
+
+        /// <summary>
+        /// For detectors loaded from Diagnose and Solve, pass x-ms-client-principal-name to Change Analysis endpoint.
+        /// </summary>
+        private string clientPrincipalNameHeader;
+
+        private readonly string changeAnalysisEndPoint = "https://changeanalysis-dataplane-dev.azurewebsites.net/providers/microsoft.changeanalysis/";
+
+        private readonly Lazy<HttpClient> client = new Lazy<HttpClient>(() =>
+        {
+            var client = new HttpClient();
+            client.DefaultRequestHeaders.Accept.Clear();
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
```

Comment:
Is there any default timeout here

Reply:
There isnt currently. But I can add that.

Reply:
I just checked it again, there is a timeout and the value is taken from `DataProviderConstants.DefaultTimeoutInSeconds`